march_gait_selection/src/march_gait_selection/transition_gait/create_transition.py

```diff
@@ -0,0 +1,206 @@
+                old_joint_setpoints.append(old_joint.get_interpolated_setpoint(timestamp))
+                new_joint_setpoints.append(new_joint.get_interpolated_setpoint(timestamp))
+
+            old_joint.setpoints = old_joint_setpoints
+            new_joint.setpoints = new_joint_setpoints
+
+        return old_subgait, new_subgait
+
+    @staticmethod
+    def _transition_trajectory(old_subgait, new_subgait):
+        """Calculate a transition trajectory which starts at the old gait and ends with the endpoints of the new gait.
+
+        :param old_subgait:
+            The subgait which the transition is starting from
+        :param new_subgait:
+            The subgait which the transition will result in.
+
+        :returns:
+             list of joints, velocities and positions corresponding to every joint at given timestamps and duration
+        """
+        joints = []
+        transition_positions = []
+        transition_velocities = []
+        transition_durations = []
+
+        total_transition_factor = len(new_subgait[0].setpoints)
+        for transition_index in range(total_transition_factor):
+            rospy.logwarn('transition index {ti} with timestamp {ts}'
+                          .format(ti=str(transition_index), ts=str(new_subgait[0][transition_index].time)))
+
+            positions = []
+            velocities = []
+            for old_joint in old_subgait.joints:
+                new_joint = new_subgait.get_joint(old_joint.name)
+
+                if len(old_joint) != len(new_joint):
+                    rospy.logerr('Amount of setpoints is inconsistent between subgait {og} of gait {fg} and {sg}'
+                                 .format(og=old_subgait.subgait_name, fg=old_subgait.gait_name,
+                                         sg=new_subgait.gait_name))
+
+                joints.append(new_joint.name)
+
+                new_setpoint_factor = 1.0 / (total_transition_factor - 1) * transition_index
+                old_setpoint_factor = 1.0 - new_setpoint_factor
+
+                old_setpoint = old_joint[transition_index]
+                new_setpoint = new_joint[transition_index]
+
+                positions.append((old_setpoint.position * old_setpoint_factor)
+                                 + (new_setpoint.position * new_setpoint_factor))
+                velocities.append((old_setpoint.velocity * old_setpoint_factor)
+                                  + (new_setpoint.velocity * new_setpoint_factor))
+
+                trans_pos = (old_setpoint.position * old_setpoint_factor) + \
+                            (new_setpoint.position * new_setpoint_factor)
+
+                rospy.logwarn('old factor: {of} \t new factor {nf} \t old: {op} \t transition: {tp} \t new: {np} '
+                              .format(of=str(old_setpoint_factor), nf=str(new_setpoint_factor),
+                                      op=str(old_setpoint.position), tp=str(trans_pos), np=str(new_setpoint.position)))
```

Comment:
Why this warning? Why only for position not for velocity? You should just use positions[-1] instead of recalculating with trans_pos, reduces the duplicate code and makes this less error prone :)
```suggestion
                                      op=str(old_setpoint.position), tp=str(positions[-1]), np=str(new_setpoint.position)))
```

Reply:
This warning is deprecated, I was using it to get some information in the logfiles